### ISSUE TITLE
Sphinx failed to build on 10.12 when using --with-mysql

### ISSUE THREAD

**Issue by sjdaws** [Homebrew/homebrew-core#3474]
Gist here: https://gist.github.com/anonymous/de835250e3c63fac6818daa81a7dec4a

Can successfully install using --interactive and configuring:
```
./configure --with-mysql-libs=/usr/local/include/mysql/mysql/ --with-mysql
```

**Comment by mistydemeo** [comment]
I can't reproduce. It looks like the problem is that it's unable to link to openssl when looking for mysql:

```
configure:7254: clang -o conftest -fno-rtti -Wall -g -D_FILE_OFFSET_BITS=64 -O3 -DNDEBUG -I/usr/local/opt/mysql/include/mysql -fno-omit-frame-pointer  -I/usr/local/include  conftest.c -lz  -L/usr/local/lib  -lpthread -L/usr/local/opt/mysql/lib -lmysqlclient -lssl -lcrypto >&5
ld: library not found for -lssl
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure:7254: $? = 1
```

Can you try reinstalling openssl?

**Comment by sjdaws** [comment]
Yeah it might not be sphinx related, I removed brew and reinstalled and cmake won't build due to errors with clang. I might just have to leave it until 10.12 is out of beta.

**Comment by sjdaws** [comment]
This seems to be an issue with cmake, I can't recompile it using `brew install cmake` or `brew install --HEAD cmake`:

https://gist.github.com/anonymous/cc09131130e860496bba57ff269fbce6

If I run it with interactive, it fails due to both gmake and clang not being able to compile a simple program throwing the error "Resource temporarily unavailable". If I chmod them inside /usr/local/Library/Homebrew/shims/super to 755 they compiler warnings stop but then I get the following error:
```
Try: clang
Line: clang  cmake_bootstrap_29559_test.c -o cmake_bootstrap_29559_test
----------  file   -----------------------

#ifdef __cplusplus
# error "The CMAKE_C_COMPILER is set to a C++ compiler"
#endif

#include<stdio.h>

#if defined(__CLASSIC_C__)
int main(argc, argv)
  int argc;
  char* argv[];
#else
int main(int argc, char* argv[])
#endif
{
  printf("%d%c", (argv != 0), (char)0x0a);
  return argc-1;
}

------------------------------------------
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Test failed to compile
```

I can't seem to get past this.

**Comment by sjdaws** [comment]
This is resolved by doing a clean install of 10.12, I've tested with both command line tools for 10.11 and 10.12 and it works for both. I'm not sure what the underlying issue is/was.
